helper.go

```diff
@@ -63,3 +64,30 @@ func decodeResponseToError(buffer []byte, msg string) error {
 	}
 	return errors.Errorf("%s: %s", msg, string(buffer))
 }
+
+// backOffConfiguration holds initial values for the initialization of a backoff that can
+// be used in retries.
+type backOffConfiguration struct {
+	// Whether to retry failed request or not.
+	Retry bool
+	// The initial (minimal) retry interval used for the exponential backoff.
+	InitialRetryInterval time.Duration
+	// MaxRetryInterval the maximum retry interval.
+	MaxRetryInterval time.Duration
+	// MaxElapsedTime is the maximum time spent on retries.
+	MaxElapsedTime time.Duration
+}
+
+// createBackOff initializes a new backoff from configured parameters.
+func (rc *backOffConfiguration) createBackOff() backoff.BackOff {
```

Comment:
`note` when we call this function it looks like: `s.backOffConf.createBackOff()`
and I don't like much when **backoff** repeated without any need.
Can we rename this function to **new** or just **create**